docs/interfaces.md

```diff
@@ -0,0 +1,571 @@
+
+        // msp 
+        X509_STORE* root_certs_orderer, // root cert store for orderer org
+        X509_STORE* root_certs_apps, // root cert store for application orgs
+
+        // tlcc session
+        session_keys_t session_keys,
+};
+
+typedef struct kvs_t {};
+
+typedef struct session_keys_t {};
+
+// TLCC contains some attestation module that is responsable to verify
+// FPC chaincode attestations.  For instance, an attestation module for
+// EPID-based attestation maintains the IAS Verification Key.
+// 
+// For instance ...
+static const char* INTEL_PUB_PEM, // Intel IAS verification key
+```
+
+# ECC
+
+## Interface:
+The ECC interface is implemented through a "normal" chaincode interface. Methods with leading underscores
+are treated as FPC commands. Normal `invoke` invocations are forwarded to a FPC chaincode enclave.
+
+```go
+// chaincode interface (exposed to admin/clients) implemented by invoke
+func Invoke(stub shim.ChaincodeStubInterface) pb.Response {}
+
+// functions below are dispatched by Invoke implementation
+
+
+// triggered by an admin
+func initEnclave(params CC_Parameters) (Credentials, error) {}
+
+// key generatation 
+func generateCCKeys(params CC_Parameters) (SignedCCKeyRegistrationMessage, error) {}
+
+// key distribution
+func exportCCKeys(credentals Credentials) (SignedExportMessage, error) {}
+func importCCKeys(params CC_Parameters) (SignedCCKeyRegistrationMessage, error) {}
+
+// chaincode invoke
+func chaincodeInvoke(transactionProposal []byte, ctx interface{}) (response, error) {}
+```
+
+```go 
+// Storage interface
+// TODO how to store sealedState, where and under what identifier (e.g. chaincode_id);
+load(chaincode_id string) (sealedState []byte, err error)
+store(chaincode_id string, sealedState []byte) error
+```
+
+This interface is implemented by ECC to let a chaincode enclave call into the peer
+
+```c++
+// Provided/implemented by a common logging module
+// interface exposed to TLCC_enclave via cgo
+void ocall_print_string([in, string] const char *str);
+```
+
+```c++
+// state access
+// TBD does this key already contain the chaincode_id/namespace?
+void ocall_get_state(
+        [in, string] const char *key, 
+        [out, size=max_val_len] uint8_t *val, uint32_t max_val_len, [out] uint32_t *val_len,
+        [in, out] sgx_cmac_128bit_tag_t *cmac,
+        [user_check] void *u_shim_ctx);
+
+void ocall_put_state(
+        [in, string] const char *key,
+        [in, size=val_len] uint8_t *val, uint32_t val_len,
+        [user_check] void *u_shim_ctx);
+
+void ocall_get_state_by_partial_composite_key(
+        [in, string] const char *comp_key,
+        [out, size=max_len] uint8_t *values, uint32_t max_len, [out] uint32_t *values_len,
+        [in, out] sgx_cmac_128bit_tag_t *cmac,
+        [user_check] void *u_shim_ctx);
+
+```
+
+```c++
+// verify msp with the help of tlcc
+void ocall_tlcc_request();
+```
+
+```c++
+// Provided/implemented by a common sealed storage module
```

Comment:
We currently have one. If created a corresponding issue #421 for that, you probably should reference that and eventually put details on internals there?